Fails to build against actix_web 4.0.0-beta.14

After trying to update to beta.14, I get the following error. The second one looks like a quick fix. Not so sure about the first one.

```
error[E0599]: no method named `extensions` found for reference `&RequestHead` in the current scope
  --> /home/.cargo/git/checkouts/actix-web-grants-fc366f5a984b4e1f/f2f9964/src/guards.rs:44:14
   |
44 |             .extensions()
   |              ^^^^^^^^^^ method not found in `&RequestHead`

error[E0599]: no method named `extensions` found for struct `HttpRequest` in the current scope
  --> /home/.cargo/git/checkouts/actix-web-grants-fc366f5a984b4e1f/f2f9964/src/permissions/mod.rs:97:17
   |
97 |             req.extensions()
   |                 ^^^^^^^^^^ method not found in `HttpRequest`
   |
  ::: /home/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-http-3.0.0-beta.15/src/http_message.rs:29:8
   |
29 |     fn extensions(&self) -> Ref<'_, Extensions>;
   |        ---------- the method is available for `HttpRequest` here
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
16 | use actix_web::HttpMessage;
```

I have reverted to an old Cargo.lock file which I'll stay with for now.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "actix-web-grants"
-version = "3.0.0-beta.4"
+version = "3.0.0-beta.5"
 authors = ["DDtKey <ddttkey@gmail.com>"]
 description = "Extension for `actix-web` to validate user permissions"
 repository = "https://github.com/DDtKey/actix-web-grants"
@@ -20,8 +20,8 @@ default = ["macro-check"]
 macro-check = ["actix-grants-proc-macro"]
 
 [dependencies]
-actix-web = { version = "4.0.0-beta.11", default-features = false }
-actix-grants-proc-macro = { path = "./proc-macro", version = "2.0.0-beta.4", optional = true }
+actix-web = { version = "4.0.0-beta.18", default-features = false }
+actix-grants-proc-macro = { path = "./proc-macro", version = "2.0.0-beta.5", optional = true }
 
 [dev-dependencies]
 actix-rt = "2"

--- a/examples/jwt-httpauth/Cargo.toml
+++ b/examples/jwt-httpauth/Cargo.toml
@@ -8,9 +8,9 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-actix-web = { version = "4.0.0-beta.11" }
+actix-web = { version = "4.0.0-beta.18" }
 actix-web-grants = { path = "../../" }
-actix-web-httpauth = "0.6.0-beta.2"
+actix-web-httpauth = "0.6.0-beta.7"
 jsonwebtoken = "7"
 serde = {version = "1.0", features = ["derive"] }
 chrono = "0.4.19"

--- a/proc-macro/Cargo.toml
+++ b/proc-macro/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "actix-grants-proc-macro"
-version = "2.0.0-beta.4"
+version = "2.0.0-beta.5"
 authors = ["DDtKey <ddttkey@gmail.com>"]
 description = "A proc-macro way to validate user permissions for `actix-web-grants` crate."
 repository = "https://github.com/DDtKey/actix-web-grants"
@@ -19,5 +19,5 @@ proc-macro2 = "1.0"
 syn = { version = "1.0", features = ["full", "derive", "extra-traits"] }
 
 [dev-dependencies]
-actix-web = { version = "4.0.0-beta.11", default-features = false }
+actix-web = { version = "4.0.0-beta.18", default-features = false }
 actix-web-grants =  { path = "../" }

--- a/src/guards.rs
+++ b/src/guards.rs
@@ -1,6 +1,5 @@
 use crate::permissions::{AuthDetails, PermissionsCheck};
-use actix_web::dev::RequestHead;
-use actix_web::guard::Guard;
+use actix_web::guard::{Guard, GuardContext};
 
 /// Implementation of Guard trait for validate permissions
 /// ```
@@ -39,9 +38,9 @@ impl PermissionGuard {
 }
 
 impl Guard for PermissionGuard {
-    fn check(&self, request: &RequestHead) -> bool {
+    fn check(&self, request: &GuardContext) -> bool {
         request
-            .extensions()
+            .req_data()
             .get::<AuthDetails>()
             .filter(|details| details.has_permission(self.allow_permission.as_str()))
             .is_some()

--- a/src/permissions/attache.rs
+++ b/src/permissions/attache.rs
@@ -12,7 +12,7 @@ use actix_web::HttpMessage;
 /// use actix_web_grants::permissions::AttachPermissions;
 /// use actix_web::dev::ServiceRequest;
 ///
-/// fn attach(req: ServiceRequest, permissions: Vec<String>) {
+/// fn attach(req: &ServiceRequest, permissions: Vec<String>) {
 ///     req.attach(permissions);
 /// }
 ///

--- a/src/permissions/mod.rs
+++ b/src/permissions/mod.rs
@@ -13,9 +13,9 @@
 //! [`AttachPermissions`]: AttachPermissions
 //! [`GrantsMiddleware`]: actix_web_grants::GrantsMiddleware;
 
-use actix_web::dev::{Payload, PayloadStream};
+use actix_web::dev::Payload;
 use actix_web::error::ErrorUnauthorized;
-use actix_web::{Error, FromRequest, HttpRequest};
+use actix_web::{Error, FromRequest, HttpMessage, HttpRequest};
 use std::future::Future;
 use std::pin::Pin;
 
@@ -90,7 +90,7 @@ impl FromRequest for AuthDetails {
     type Error = Error;
     type Future = Pin<Box<dyn Future<Output = Result<Self, Error>>>>;
 
-    fn from_request(req: &HttpRequest, _payload: &mut Payload<PayloadStream>) -> Self::Future {
+    fn from_request(req: &HttpRequest, _payload: &mut Payload) -> Self::Future {
         let req = req.clone();
 
         Box::pin(async move {

--- a/tests/common.rs
+++ b/tests/common.rs
@@ -1,6 +1,6 @@
 use actix_web::dev::{ServiceRequest, ServiceResponse};
 use actix_web::error::ErrorUnauthorized;
-use actix_web::http::{header::AUTHORIZATION, HeaderValue};
+use actix_web::http::header::{HeaderValue, AUTHORIZATION};
 use actix_web::{test, Error};
 use serde::Deserialize;
 

